docs/httpproxy.md

```diff
@@ -0,0 +1,833 @@
+    - `retryPolicy.perTryTimeout` specifies the timeout per retry. If this field is greater than the request timeout, it is ignored. This parameter is optional. 
+    If left unspecified, `timeoutPolicy.request` will be used. 
+
+#### Load Balancing Strategy
+
+Each upstream service can have a load balancing strategy applied to determine which of its Endpoints is selected for the request.
+The following list are the options available to choose from:
+
+- `RoundRobin`: Each healthy upstream Endpoint is selected in round robin order (Default strategy if none selected).
+- `WeightedLeastRequest`: The least request strategy uses an O(1) algorithm which selects two random healthy Endpoints and picks the Endpoint which has fewer active requests. Note: This algorithm is simple and sufficient for load testing. It should not be used where true weighted least request behavior is desired.
+- `Random`: The random strategy selects a random healthy Endpoints.
+
+More information on the load balancing strategy can be found in [Envoy's documentation](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/load_balancing.html).
+
+The following example defines the strategy for Service `s2-strategy` as `WeightedLeastRequest`.
+Service `s1-strategy` does not have an explicit strategy defined so it will use the strategy of `RoundRobin`.
+
+```yaml
+# lb-strategy.httpproxy.yaml
+apiVersion: projectcontour.io/v1alpha1
+kind: HTTPProxy
+metadata:
+  name: lb-strategy
+  namespace: default
+spec:
+  virtualhost:
+    fqdn: strategy.bar.com
+  routes:
+    - conditions:
+        prefix: /
+      services:
+        - name: s1-strategy
+          port: 80
+        - name: s2-strategy
+          port: 80
+          strategy: WeightedLeastRequest
+```
+#### Session Affinity
+
+Session affinity, also known as _sticky sessions_, is a load balancing strategy whereby a sequence of requests from a single client are consitently routed to the same application backend.
+Contour supports session affinity with the `strategy: Cookie` key on a per service basis.
+
+```yaml
+apiVersion: projectcontour.io/v1alpha1
+kind: HTTPProxy
+metadata:
+  name: httpbin
+  namespace: default
+spec:
+  virtualhost:
+    fqdn: httpbin.davecheney.com
+  routes:
+  - conditions:
+      prefix: /
+    services:
+    - name: httpbin
+      port: 8080
+      strategy: Cookie
+```
+##### Limitations
+
+Session affinity is based on the premise that the backend servers are robust, do not change ordering, or grow and shrink according to load.
+None of these properties are guaranteed by a Kubernetes cluster and will be visible to applications that rely heavily on session affinity.
+
+Any pertibation in the set of pods backing a service risks redistributing backends around the hash ring.
+This is an unavoidable consiquence of Envoy's session affinity implementation and the pods-as-cattle approach of Kubernetes.
+
+#### Per-Upstream Active Health Checking
```

Comment:
We should probably have a note about how healthchecks work with multiple upstreams and the dangers involved.